### ISSUE TITLE
Nodes of RKE2 cluster not deleted from backend when cluster is deleted

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#32868]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
On master-head commit id: `f90b36e`

- Create clusters  2 RKE2 DO clusters created from cluster explorer and 1 RKE1 cluster created from Ember UI
- When the clusters are up and running, delete the clusters by selecting them all from Ember UI.
- The clusters get deleted
- The nodes of RKE2 clusters are _not_ deleted from backend
- The nodes of the RKE1 cluster _is_ deleted

**Expected Result:**
Nodes should be deleted from Digital Ocean Backend

**Other details that may be helpful:**
Rancher logs:
```
2021/05/25 18:59:53 [INFO] Deleting cluster [c-m-facc49ca]
2021/05/25 18:59:53 [INFO] [mgmt-cluster-rbac-remove] Deleting namespace c-m-facc49ca
2021/05/25 18:59:53 [INFO] [mgmt-auth-users-controller] Deleting clusterRoleTemplateBinding u-3fc77pwz6h-admin for user u-3fc77pwz6h
2021/05/25 18:59:54 [INFO] [mgmt-auth-users-controller] Deleting globalRoleBinding grb-dbbb7 for user u-3fc77pwz6h
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-d2kmxgqmq7
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-3fc77pwz6h-admin-cluster-owner in namespace p-thj5j for crtb u-3fc77pwz6h-admin
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca': handler mgmt-cluster-remove: clusters.provisioning.cattle.io "sowmya-2" not found, requeuing
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-3fc77pwz6h-admin-cluster-owner in namespace p-9j8q5 for crtb u-3fc77pwz6h-admin
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Updating owner label for roleBinding crb-smjcbfenrk
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-m-facc49ca-owner-cluster-owner in namespace p-8276l for crtb c-m-facc49ca-owner
2021/05/25 18:59:54 [INFO] Deleting cluster [c-m-867d20db]
2021/05/25 18:59:54 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-m-facc49ca-owner-cluster-owner in namespace p-26s6v for crtb c-m-facc49ca-owner
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] Error deleting resource: error deleting cluster fleet-default/sowmya-2: failed to delete cluster.x-k8s.io/v1alpha4, Kind=MachineDeployment sowmya-2-nodepool-pool1: machinedeployments.cluster.x-k8s.io "sowmya-2-nodepool-pool1" not found
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca': handler user-controllers-controller: clusters.management.cattle.io "c-m-facc49ca" not found, handler mgmt-cluster-remove: Operation cannot be fulfilled on clusters.management.cattle.io "c-m-facc49ca": StorageError: invalid object, Code: 4, Key: /registry/management.cattle.io/clusters/c-m-facc49ca, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 7d1c804b-7001-4d38-886b-a4bcc54fcaab, UID in object meta: , requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] Reconciler error: error deleting cluster fleet-default/sowmya-2: failed to delete cluster.x-k8s.io/v1alpha4, Kind=MachineDeployment sowmya-2-nodepool-pool1: machinedeployments.cluster.x-k8s.io "sowmya-2-nodepool-pool1" not found
2021/05/25 18:59:54 [INFO] [mgmt-cluster-rbac-remove] Deleting namespace c-m-867d20db
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:54 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:55 [INFO] [mgmt-auth-crtb-controller] Updating owner label for roleBinding crb-hhrpl2acuz
2021/05/25 18:59:55 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-m-867d20db-owner-cluster-owner in namespace p-thj5j for crtb c-m-867d20db-owner
2021/05/25 18:59:55 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-m-867d20db-owner-cluster-owner in namespace p-9j8q5 for crtb c-m-867d20db-owner
2021/05/25 18:59:55 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
2021/05/25 18:59:55 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:55 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:55 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:55 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:56 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
2021/05/25 18:59:56 [ERROR] Reconciler error: clusters.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] Reconciler error: failed to get cluster "sowmya-2" for machine "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] Reconciler error: failed to get cluster "sowmya-2" for machine "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] Reconciler error: failed to get cluster "sowmya-2" for machine "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] Reconciler error: failed to get cluster "sowmya-2" for machine "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] Reconciler error: failed to get cluster "sowmya-2" for machine "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-2" not found
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:56 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:57 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:57 [ERROR] error syncing 'fleet-default/sowmya-2-nodepool-pool1-ctj95': handler machine-provision: Operation cannot be fulfilled on digitaloceanmachines.rke-node.cattle.io "sowmya-2-nodepool-pool1-ctj95": StorageError: invalid object, Code: 4, Key: /registry/rke-node.cattle.io/digitaloceanmachines/fleet-default/sowmya-2-nodepool-pool1-ctj95, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 4ce7fa39-3949-4502-a83c-c90ccc797f07, UID in object meta: , requeuing
2021/05/25 18:59:57 [ERROR] error syncing 'c-m-facc49ca/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:57 [ERROR] error syncing 'c-m-facc49ca/system': handler cluster-registration-token: clusters.management.cattle.io "c-m-facc49ca" not found, requeuing
2021/05/25 18:59:57 [ERROR] error syncing 'fleet-default/sowmya-2-bootstrap-template-wmqdx': handler rke-machine: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-2-nodepool-pool1-599f5d8d6b-sh4dt" not found, requeuing
2021/05/25 18:59:58 [ERROR] Reconciler error: clusters.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] Reconciler error: failed to get cluster "sowmya-1" for machine "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] Reconciler error: failed to get cluster "sowmya-1" for machine "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] Reconciler error: failed to get cluster "sowmya-1" for machine "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] Reconciler error: failed to get cluster "sowmya-1" for machine "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] Reconciler error: failed to get cluster "sowmya-1" for machine "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" in namespace "fleet-default": Cluster.cluster.x-k8s.io "sowmya-1" not found
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-nodepool-pool1-p8jvk': handler machine-provision: Operation cannot be fulfilled on digitaloceanmachines.rke-node.cattle.io "sowmya-1-nodepool-pool1-p8jvk": StorageError: invalid object, Code: 4, Key: /registry/rke-node.cattle.io/digitaloceanmachines/fleet-default/sowmya-1-nodepool-pool1-p8jvk, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: f0a607ea-9c12-4bbf-8e46-06c4be9acba3, UID in object meta: , requeuing
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:58 [INFO] kontainerdriver rancherkubernetesengine listening on address 127.0.0.1:46395
2021/05/25 18:59:58 [INFO] cluster [c-bcmlx] provisioning: Tearing down Kubernetes cluster
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:58 [INFO] cluster [c-bcmlx] provisioning: [dialer] Setup tunnel for host [64.225.15.152]
2021/05/25 18:59:58 [INFO] cluster [c-bcmlx] provisioning: [worker] Tearing down Worker Plane..
2021/05/25 18:59:58 [INFO] Removing container [kubelet] on host [64.225.15.152], try       #1
2021/05/25 18:59:58 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
2021/05/25 18:59:59 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-smjcbfenrk
2021/05/25 18:59:59 [ERROR] error syncing 'fleet-default/sowmya-1-bootstrap-template-9ktsq': handler rke-machine: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, handler machine-node-lookup: machines.cluster.x-k8s.io "sowmya-1-nodepool-pool1-656fdf859c-bfp2m" not found, requeuing
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):  master-head commit id: `f90b36e`
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Do RKE2
- Kubernetes version (use `kubectl version`):

```
1.21.1-alpha5-rke2r1
```

**Comment by deniseschannon** [comment]
There were some issues around invoking the machine driver to delete the machines that may have now been corrected, but the code within machine is still being used to delete the backend machines

**Comment by thehejik** [comment]
**Test report:**
Tested on `rancher/rancher:master-head 97ca638` - on local RKE1 cluster (on k8s v1.20.7) - I've created 2 DO RKE2 clusters from Explorer and 1 DO RKE1 cluster from Ember.

One RKE1 (3 nodes with all roles on k8s v1.20.7) and two RKE2 (3 nodes with all roles on k8s v1.21.1 each) clusters were deleted at once in Ember - also all nodes were destroyed automatically and removed from DO.

This shouldn't have a backport into v2.5 as RKE2 provisioning is available only in Explorer. Close
